tests/test_client_functional.py

```diff
@@ -2166,6 +2166,35 @@ async def handler_redirect(request):
     assert data == body
 
 
+@pytest.mark.parametrize(
+    "invalid_redirect_url",
+    (
+        # yarl.URL.__new__ raises ValueError
+        "http://:/",
+        "http://www.example.com/../filename.html",
+        "http://example.org:non_int_port/",
+        # yarl.URL.origin raises ValueError
+        "http:/",
+        "http:/example.com",
+        "http:///example.com",
```

Comment:
Let's also test schemas used to invoke custom non-browser apps:
```suggestion
        "http:///example.com",
        "bluesky://d:i:d",
        "call:+3801234567",
        "skype:handle",
        "slack://instance/room",
        "steam:code",
        "twitter://handle",
```

Reply:
added